src/SymfonyCache/UserContextSubscriber.php

```diff
@@ -180,6 +180,15 @@ private function getUserHash(HttpKernelInterface $kernel, Request $request)
      */
     private function isAnonymous(Request $request)
     {
+        // You might have to enable rewriting of the Authorization header in your server config or .htaccess:
+        // RewriteEngine On
+        // RewriteRule .* - [E=HTTP_AUTHORIZATION:%{HTTP:Authorization}]
+        foreach ($request->server as $name => $value) {
```

Comment:
lets do `if ($request->server->has('HTTP_AUTHORIZATION') ...` rather than loop through all the headers.

looking at the ServerBag::getHeaders method, i have the impression that the safe thing to do would be to check for any of the headers AUTHORIZATION, HTTP_AUTHORIZATION or PHP_AUTH_USER and consider the request not anonymous if any of those headers is present.